src/disk_io_thread.cpp

```diff
@@ -2792,12 +2792,20 @@ namespace libtorrent
 		cached_piece_entry* pe = m_disk_cache.find_piece(j);
 		if (pe == nullptr) return 0;
 
+		piece_refcount_holder refcount_holder(pe);
+
 #if TORRENT_USE_ASSERTS
 		pe->piece_log.push_back(piece_log_t(j->action));
 #endif
 		try_flush_hashed(pe, m_settings.get_int(
```

Comment:
``do_flush_hashed()`` already increments the refcount while flushing, this shouldn't be necessary.